db/database.py

```diff
@@ -0,0 +1,192 @@
+
+    def add_rooms(self):
+        """Add data from the Rooms list to the database"""
+        rooms = Table(
+                "rooms", self.metadata,
+                Column("name", String, primary_key=True),
+                Column("is_office", Boolean),
+                Column("is_vacant", Boolean),
+                extend_existing=True
+        )
+        if rooms.exists():
+            rooms.update()
+        else:
+            rooms.create()
+        i = rooms.insert()
+        for room in self.my_amity.rooms:
+            if room not in self.already_added:
+                if room.room_type == "Office":
+                    is_office = True
+                else:
+                    is_office = False
+                if room in self.my_amity.vacant_rooms:
+                    is_vacant = True
+                else:
+                    is_vacant = False
+                i.execute(
+                    name=room.name,
+                    is_office=is_office,
+                    is_vacant=is_vacant
+                )
+                self.already_added.append(room)
+
+    def add_allocations(self):
+        """
+        Add data from the Allocations list to the database.
+        Each room gets a database table with occupants as rows.
+        """
+        if self.my_amity.rooms:
+            for room in self.my_amity.rooms:
+                room_name = room.name
+                room_name = Table(
+                        room_name, self.metadata,
+                        Column(
+                            "employee_id", Integer,
+                            ForeignKey("people.employee_id"),
+                            nullable=False),
+                        Column("name", String),
+                        extend_existing=True
+                )
+                if room_name.exists():
+                    room_name.update()
+                else:
+                    room_name.create()
+                i = room_name.insert()
+                if room.occupants:
+                    for person in room.occupants:
+                        i.execute(
+                            employee_id=person.emp_id,
+                            name=person.name
+                        )
+
+    def load_state(self, args):  # pragma: no cover
+        """Load data to the application from a user-defined database"""
+        self.db_name = "sqlite:///" + "%s" % args.get("<sqlite_database>")
+        self.db = create_engine(self.db_name)
+        connection = self.db.connect()
+        people_ = connection.execute("SELECT * FROM people")
+        rooms_ = connection.execute("SELECT * FROM rooms")
+        table_names = connection.execute(
+            "SELECT name FROM sqlite_master \
+            WHERE (type='table') AND (name NOT LIKE 'people') \
+            AND (name NOT LIKE 'rooms') "
+        )
+
+        for row in people_:
+            print('people in load_state', people_)
+            if row["is_fellow"]:
```

Comment:
**Code Quality: Nitpick**

Replace all instances where you did this sort of logic with single line variants. In this case `is_fellow = row['is_fellow']` would do the exact same thing. If you need to convert `None` or empty string to a boolean you can make use of the `bool()` function.